CustomCollections/CustomCollections/CustomVector.cs

```diff
@@ -0,0 +1,132 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace CustomCollections
+{
+    class CustomVector<T> : IList<T>
+    {
+        private T[] _vector;
+
+        public int Count { get; private set; }
+        public bool IsReadOnly { get; private set; } = true;
+
+        public CustomVector()
+        {
+            _vector = new T[0];
+            Count = _vector.Length;
+        }
+
+        public T this[int index]
+        {
+            get => _vector[index];
+            set => _vector[index] = value;
+        }
+
+        public void Add(T item)
+        {
+            var newVector = new T[Count + 1];
+            for (int i = 0; i < Count; i++)
+            {
+                newVector[i] = _vector[i];
+            }
+            newVector[Count] = item;
+            Count++;
+            _vector = newVector;
+            IsReadOnly = false;
+        }
+
+        public void Clear()
+        {
+            _vector = new T[0];
+            Count = 0;
+            IsReadOnly = true;
+        }
+
+        public bool Contains(T item)
+        {
+            if (_vector.Contains(item))
+            {
+                return true;
+            }
+            return false;
+        }
+
+        public bool Remove(T item)
+        {
+            _vector = _vector.Where(s => !s.Equals(item)).ToArray();
```

Comment:
This is not the way remove method works. It should remove only FIRST occurrence. Refer to: https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.icollection-1.remove?view=netframework-4.7.2#System_Collections_Generic_ICollection_1_Remove__0_

Reply:
And again, I don't want to see Linq here (Only for data structures implementation, use it everywhere else).